controllers/serviceaccount_controller.go

```diff
@@ -0,0 +1,879 @@
+		}
+
+		// Service account not found, clear the status and create a new one
+		cr.Status.Account = nil
+	}
+
+	create, err := gClient.ServiceAccounts.CreateServiceAccount(
+		service_accounts.
+			NewCreateServiceAccountParamsWithContext(ctx).
+			WithBody(&models.CreateServiceAccountForm{
+				Name:       cr.Spec.Name,
+				Role:       cr.Spec.Role,
+				IsDisabled: cr.Spec.IsDisabled,
+			}),
+	)
+	if err != nil {
+		return fmt.Errorf("creating service account: %w", err)
+	}
+
+	cr.Status.Account = &v1beta1.GrafanaServiceAccountInfo{
+		ID:         create.Payload.ID,
+		Role:       create.Payload.Role,
+		IsDisabled: create.Payload.IsDisabled,
+		Name:       create.Payload.Name,
+		Login:      create.Payload.Login,
+	}
+
+	return nil
+}
+
+func (r *GrafanaServiceAccountReconciler) removeAccountToken(
+	ctx context.Context,
+	gClient *genapi.GrafanaHTTPAPI,
+	serviceAccountID int64,
+	tokenStatus *v1beta1.GrafanaServiceAccountTokenStatus,
+) error {
+	if tokenStatus.Secret != nil {
+		secret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{
+			Namespace: tokenStatus.Secret.Namespace,
+			Name:      tokenStatus.Secret.Name,
+		}}
+
+		err := r.Delete(ctx, secret)
+		if err != nil && !kuberr.IsNotFound(err) {
+			return fmt.Errorf("deleting token secret %q: %w", secret.Name, err)
+		}
+
+		tokenStatus.Secret = nil
+	}
+
+	_, err := gClient.ServiceAccounts.DeleteTokenWithParams( // nolint:errcheck
+		service_accounts.
+			NewDeleteTokenParamsWithContext(ctx).
+			WithServiceAccountID(serviceAccountID).
+			WithTokenID(tokenStatus.ID),
+	)
+	if err != nil {
+		// ATM, service_accounts.DeleteTokenNotFound doesn't have Is, Unwrap, Unwrap.
+		// So, we cannot rely only on errors.Is().
+		_, ok := err.(*service_accounts.DeleteTokenNotFound) // nolint:errorlint
+		if ok || errors.Is(err, service_accounts.NewDeleteTokenNotFound()) {
+			return nil
+		}
+
+		return fmt.Errorf("deleting token: %w", err)
+	}
+
+	return nil
+}
+
+func isEqualExpirationTime(a, b *metav1.Time) bool {
+	// Token expiration drift tolerance
+	const tokenExpirationDrift = 1 * time.Second
+
+	if a == nil && b == nil {
+		return true
+	}
+
+	if a == nil || b == nil {
+		return false
+	}
+
+	// Grafana API doesn't allow to set expiration time for tokens. Instead of it,
+	// Grafana accepts TTL then calculates the expiration time against the current time.
+	// So, we cannot just compare the expiration time with the spec' one.
+	// Let's assume that two expiration times are equal if they are close enough.
+	diff := a.Sub(b.Time)
+
+	return diff.Abs() <= tokenExpirationDrift
+}
+
+func buildSecretLabels(cr *v1beta1.GrafanaServiceAccount) map[string]string {
+	return map[string]string{
+		"app": "grafana-serviceaccount-token",
+		"grafana.integreatly.org/service-account-instance": cr.Spec.InstanceName,
+		"grafana.integreatly.org/service-account-name":     cr.Name,
+		"grafana.integreatly.org/service-account-uid":      string(cr.UID),
```

Comment:
`integreatly` is just an historical artifact. We can do `operator.grafana.com` instead (as we do with the finalizer)